### ISSUE TITLE
💅 noUnnecessaryConditions incorrectly flags a switch condition

### ISSUE THREAD

**Issue by lawfulsoftware** [biomejs/biome#8332]
### Environment information

```bash
CLI:
  Version:                      2.3.8
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            ~/coreutils/biome.json
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v21.7.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         ~/coreutils/biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 unset
  CSS enabled:                  true
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementInteractions
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noEmptyTypeParameters
    complexity/noExcessiveCognitiveComplexity
    complexity/noExcessiveLinesPerFunction
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noForEach
    complexity/noImplicitCoercions
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringConcat
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/noVoid
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    complexity/useSimplifiedLogicExpression
    complexity/useWhile
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalDirnameFilename
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNestedComponentDefinitions
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noPrivateImports
    correctness/noReactPropAssignments
    correctness/noRenderReturnValue
    correctness/noRestrictedElements
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noSolidDestructuredProps
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredDependencies
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useHookAtTopLevel
    correctness/useIsNan
    correctness/useJsonImportAttributes
    correctness/useJsxKeyInIterable
    correctness/useParseIntRadix
    correctness/useSingleJsDocAsterisk
    correctness/useUniqueElementIds
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noDeprecatedImports
    nursery/noDuplicateDependencies
    nursery/noEmptySource
    nursery/noFloatingPromises
    nursery/noForIn
    nursery/noImportCycles
    nursery/noJsxLiterals
    nursery/noMisusedPromises
    nursery/noMultiStr
    nursery/noParametersOnlyUsedInRecursion
    nursery/noProto
    nursery/noShadow
    nursery/noUnnecessaryConditions
    nursery/noUnusedExpressions
    nursery/noUselessCatchBinding
    nursery/noUselessUndefined
    nursery/useArraySortCompare
    nursery/useConsistentArrowReturn
    nursery/useExhaustiveSwitchCases
    nursery/useFind
    nursery/useMaxParams
    nursery/useSortedClasses
    nursery/useSpread
    performance/noAccumulatingSpread
    performance/noAwaitInLoops
    performance/noBarrelFile
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noImgElement
    performance/noNamespaceImport
    performance/noReExportAll
    performance/noUnwantedPolyfillio
    performance/useGoogleFontPreconnect
    performance/useSolidForComponent
    performance/useTopLevelRegex
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noCommonJs
    style/noDefaultExport
    style/noDescendingSpecificity
    style/noDoneCallback
    style/noEnum
    style/noExportedImports
    style/noHeadElement
    style/noImplicitBoolean
    style/noInferrableTypes
    style/noMagicNumbers
    style/noNamespace
    style/noNegationElse
    style/noNestedTernary
    style/noNonNullAssertion
    style/noParameterAssign
    style/noParameterProperties
    style/noProcessEnv
    style/noRestrictedGlobals
    style/noRestrictedImports
    style/noRestrictedTypes
    style/noShoutyConstants
    style/noSubstr
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/noValueAtRule
    style/noYodaExpression
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useAtIndex
    style/useCollapsedElseIf
    style/useCollapsedIf
    style/useComponentExportOnlyModules
    style/useConsistentArrayType
    style/useConsistentBuiltinInstantiation
    style/useConsistentCurlyBraces
    style/useConsistentMemberAccessibility
    style/useConsistentObjectDefinitions
    style/useConsistentTypeDefinitions
    style/useConst
    style/useDefaultParameterLast
    style/useDefaultSwitchClause
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExponentiationOperator
    style/useExportType
    style/useFilenamingConvention
    style/useForOf
    style/useFragmentSyntax
    style/useGroupedAccessorPairs
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodeAssertStrict
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useNumericSeparators
    style/useObjectSpread
    style/useReadonlyClassProperties
    style/useSelfClosingElements
    style/useShorthandAssign
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useSymbolDescription
    style/useTemplate
    style/useThrowNewError
    style/useThrowOnlyError
    style/useTrimStartEnd
    style/useUnifiedTypeSignatures
    suspicious/noAlert
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noBitwiseOperators
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConsole
    suspicious/noConstEnum
    suspicious/noConstantBinaryExpressions
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDocumentImportInPage
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyBlockStatements
    suspicious/noEmptyInterface
    suspicious/noEvolvingTypes
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFocusedTests
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noHeadImportInDocument
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisplacedAssertion
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noReactSpecificProps
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSkippedTests
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnassignedVariables
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useAwait
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useErrorMessage
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useGuardForIn
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword
    suspicious/useNumberToFixedDigitsArgument
    suspicious/useStaticResponseMethods
    suspicious/useStrictMode

Workspace:
  Open Documents:               0
```

### Rule name

lint/nursery/noUnnecessaryConditions

### Playground link

https://biomejs.dev/playground/?lintRules=noUnnecessaryConditions&code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIAByAGUAcABsAGEAYwBlAHIAKAAgAF8AawBlAHkAOgAgAHMAdAByAGkAbgBnACwAIAB2AGEAbAB1AGUAOgAgAHUAbgBrAG4AbwB3AG4AIAApACAAewAKAAkAYwBvAG4AcwB0ACAAdAB5AHAAZQAgAD0AIAB0AHkAcABlAG8AZgAgAHYAYQBsAHUAZQA7AAoACQBzAHcAaQB0AGMAaAAgACgAIAB0AHkAcABlACAAKQAgAHsACgAJAAkAYwBhAHMAZQAgACcAbgB1AG0AYgBlAHIAJwA6AAoACQAJAGMAYQBzAGUAIAAnAHMAdAByAGkAbgBnACcAOgAKAAkACQBjAGEAcwBlACAAJwBiAG8AbwBsAGUAYQBuACcAOgAKAAkACQBjAGEAcwBlACAAJwBiAGkAZwBpAG4AdAAnADoACgAJAAkACQByAGUAdAB1AHIAbgAgAHQAeQBwAGUAOwAKAAkACQBjAGEAcwBlACAAJwBvAGIAagBlAGMAdAAnADoACgAJAAkACQBpAGYAIAAoACAAdgBhAGwAdQBlACAAPQA9AD0AIABuAHUAbABsACAAKQAgAHIAZQB0AHUAcgBuACAAJwBuAHUAbABsACcAOwAKAAkACQAJAGkAZgAgACgAIABBAHIAcgBhAHkALgBpAHMAQQByAHIAYQB5ACgAIAB2AGEAbAB1AGUAIAApACAAKQAgAHIAZQB0AHUAcgBuACAAQQByAHIAYQB5AFQAbwBUAHkAcABlAHMAKAAgAHYAYQBsAHUAZQAgAGEAcwAgAHUAbgBrAG4AbwB3AG4AWwBdACAAKQA7AAoACQAJAAkAaQBmACAAKAAgAE8AYgBqAGUAYwB0AC4AawBlAHkAcwAoACAAdgBhAGwAdQBlACAAYQBzACAAUgBlAGMAbwByAGQAPABzAHQAcgBpAG4AZwAsACAAdQBuAGsAbgBvAHcAbgA%2BACAAKQAuAGwAZQBuAGcAdABoACAAPQA9AD0AIAAwACAAKQAgAHIAZQB0AHUAcgBuACAAJwBSAGUAYwBvAHIAZAA8AHMAdAByAGkAbgBnACwAIAB1AG4AawBuAG8AdwBuAD4AJwA7AAoACQAJAAkAcgBlAHQAdQByAG4AIAB2AGEAbAB1AGUAOwAKAAkACQBkAGUAZgBhAHUAbAB0ADoACgAJAAkACQB0AGgAcgBvAHcAIABuAGUAdwAgAFQAeQBwAGUARQByAHIAbwByACgAIABgAFUAbgBlAHgAcABlAGMAdABlAGQAIAB0AHkAcABlADoAIAAkAHsAdAB5AHAAZQB9AFwAbgAkAHsAdgBhAGwAdQBlAH0AYAAgACkAOwAKAAkAfQAKAH0A

### Expected result

It is erroneously warning: This condition is always truthy based on the type.

if `switch ( type )` is replaced with `switch ( typeof value )`, the linter does not warn.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by JClackett** [comment]
a slightly simpler playground

https://biomejs.dev/playground/?lintRules=noUnnecessaryConditions&code=ZgB1AG4AYwB0AGkAbwBuACAAdABlAHMAdAAoAGYAbwBvADoAIAAiAGEAIgAgAHwAIAAiAGIAIgAgAHwAIAAiAGMAIgApACAAewAKACAAIABzAHcAaQB0AGMAaAAgACgAZgBvAG8AKQAgAHsACgAgACAAIAAgAGMAYQBzAGUAIAAiAGEAIgA6AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIAAxAAoAIAAgACAAIABjAGEAcwBlACAAIgBiACIAOgAKACAAIAAgACAAIAAgAHIAZQB0AHUAcgBuACAAMgAKACAAIAAgACAAYwBhAHMAZQAgACIAYwAiADoACgAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgADMACgAgACAAfQAKAH0ACgAKAHQAZQBzAHQAKAAiAGEAIgApAA%3D%3D

**Comment by ematipico** [comment]
Closing. The issues has been fixed in one of latest versions of Biome.

**Comment by jpike88** [comment]
I'm not so sure, I just ran into this issue building from source on the latest commit on main branch [6671ac5b4d](https://github.com/biomejs/biome/commit/6671ac5b4d18d2ae94932d1ae710c3a3b7ed1c4c)

**Comment by ematipico** [comment]
The playground is clean. Open a new issue if you think there's an issue

**Comment by jpike88** [comment]
I had trouble getting any result on the playground, see this link:

https://biomejs.dev/playground/?lintRules=noUnnecessaryConditions&tab=formatter&pane=Diagnostics&code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIAB0AGUAcwB0ACgAdgBhAGwAdQBlADoAIAAiAGEAIgAgAHwAIAAiAGIAIgAgAHwAIAAiAGMAIgApACAAewAKACAAIABjAG8AbgBzAHQAIAB4ACAAPQAgAHQAcgB1AGUAOwAKACAAIABpAGYAKAB4ACkAewAKACAAIAAgACAALwAvACAAPwA%2FAD8AIABpACAAZABvAG4AJwB0ACAAcwBlAGUAIABhACAAbABpAG4AdABpAG4AZwAgAGUAcgByAG8AcgAgAGgAZQByAGUAPwAKACAAIAB9AAoAIAAgAHMAdwBpAHQAYwBoACAAKAB2AGEAbAB1AGUAKQAgAHsACgAgACAAIAAgAGMAYQBzAGUAIAAiAGEAIgA6AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIAAxADsACgAgACAAIAAgAGMAYQBzAGUAIAAiAGIAIgA6AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIAAyADsACgAgACAAIAAgAGMAYQBzAGUAIAAiAGMAIgA6AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIAAzADsACgAgACAAfQAKAH0A

I have specifically enabled `noUnecessaryConditions`, and also included a simpler case. For some reason it's not picking it up... am I doing it wrong?
